edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs

```diff
@@ -292,6 +292,7 @@ class ConnectedDevice
             // so using traditional locking mechanism for those.
             readonly object deviceProxyLock = new object();
             readonly AsyncLock cloudConnectionLock = new AsyncLock();
+            Option<Task<Try<ICloudConnection>>> cloudConnectionCreateTask = Option.None<Task<Try<ICloudConnection>>>();
```

Comment:
Does this ever get set back to `None`? Or is it expected that one of the calls will return an inactive connection and then it will get refreshed?

Reply:
Right, if the connection is inactive and the task is completed, then it gets refreshed.